javagen/src/main/java/com/azure/autorest/mapper/ModelMapper.java

```diff
@@ -46,31 +47,46 @@ public ClientModel map(ObjectSchema compositeType) {
 
             String parentModel = null;
             boolean hasAdditionalProperties = false;
+            List<ObjectSchema> parentsNeedFlatten = new ArrayList<>();
             if (compositeType.getParents() != null && compositeType.getParents().getImmediate() != null) {
-                if (!(compositeType.getParents().getImmediate().get(0) instanceof DictionarySchema)) {
-//                ComplexSchema baseSchema = compositeType.getParents().getImmediate().get(0);
-//                if (baseSchema instanceof ObjectSchema) {
-//                    parentModel = map((ObjectSchema) baseSchema);
-//                    serviceModels.addModel(parentModel);
-//                } else {
-//                    throw new RuntimeException("Wait what? How? Parent is not an object but a " + baseSchema.getClass() + "?");
-//                }
-                    ComplexSchema parentComplexSchema = compositeType.getParents().getImmediate().get(0);
-                    if (parentComplexSchema instanceof ObjectSchema) {
-                        ClassType parentType = objectMapper.map((ObjectSchema) parentComplexSchema);
-                        parentModel = parentType.getName();
-                        modelImports.add(parentType.getPackage() + "." + parentModel);
-                    } else {
-                        parentModel = compositeType.getParents().getImmediate().get(0).getLanguage().getJava().getName();
+                hasAdditionalProperties = compositeType.getParents().getImmediate().stream()
+                        .anyMatch(s -> s instanceof DictionarySchema);
+
+                ObjectSchema firstParentComplexSchema = null;
+                for (Schema parent : compositeType.getParents().getImmediate()) {
+                    if (parent instanceof ObjectSchema) {
+                        if (firstParentComplexSchema == null) {
+                            firstParentComplexSchema = (ObjectSchema) parent;
+                        } else {
+                            parentsNeedFlatten.add((ObjectSchema) parent);
+                        }
                     }
```

Comment:
Some mgmt-plane swagger has type with multiple base type. Currently I only take the first one as base, and flatten the rest. Could there be a better heuristic, like take the one that itself inherits a base class or got a discriminator?

v1 flatten it as well. But not sure how it pick the base class.

Reply:
I agree the right way to handle multiple inheritance in Java is through composition. In v1 flattening was done by modeler so we didn't have to implement that.
While we can reach out to Garrett for advice, I think keeping the same behavior as v1 should be okay.

Reply:
Yes, agree. I will try composition, see if I can pull off the serialization/de-serialization correctly.

Reply:
JsonUnwrapper should do (however even when I do not add this, the serialization/de-serialization of the properties under composition still works...).

However the complete task would affect quite much to ClientModel and ModelTemplate, since the currently ModelTemplate and ClientModelProperty appears not able to handle the properties from composition.
I will delay the implementation of composition to a future task.